src/array_decoder/string.rs

```diff
@@ -133,12 +132,12 @@ impl<T: ByteArrayType> GenericByteArrayDecoder<T> {
             elements_to_fetch,
             "less lengths than expected in ByteArray"
         );
-        let total_length: u64 = lengths.iter().sum();
+        let total_length: i64 = lengths.iter().sum();
         // Fetch all data bytes at once
         let mut bytes = Vec::with_capacity(total_length as usize);
         self.bytes
             .by_ref()
-            .take(total_length)
+            .take(total_length as u64) // TODO
```

Comment:
Hi, would you like to explain what's this TODO for?

Reply:
Can't remember what I was thinking at that point; probably something to do with it's sign? But I think it's actually a non-issue as `total_length` is originally an i64 anyway, will remove the TODO :+1: